test/Microsoft.AspNetCore.Http.Tests/Internal/DefaultHttpRequestTests.cs

```diff
@@ -194,6 +195,17 @@ public void Cookies_GetAndSet()
             Assert.Equal(new[] { "name2=value2" }, cookieHeaders);
         }
 
+        [Fact]
+        public void RouteValues_GetAndSet()
```

Comment:
Add a test that verifies the feature is read from the feature collection if implemented.

```C#
var features = new FeatureCollection();
features.Set<IRouteValuesFeature>(new RouteValuesFeature() { 
   Values = new RouteValueDictionary( new { A = 1 })
 });
var context = new DefaultHttpContext(features);
Assert.Equal(1, context.Request.RouteValues["A"]);
```